plugins/destination/duckdb/client/transform.go

```diff
@@ -0,0 +1,92 @@
+package client
+
+import (
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/apache/arrow/go/v13/arrow/array"
+	"github.com/apache/arrow/go/v13/arrow/memory"
+	"github.com/cloudquery/plugin-sdk/v3/types"
+)
+
+func transformRecord(sc *arrow.Schema, rec arrow.Record) arrow.Record {
+	cols := make([]arrow.Array, rec.NumCols())
+	for i := 0; i < int(rec.NumCols()); i++ {
+		cols[i] = transformArray(rec.Column(i))
+	}
+	return array.NewRecord(sc, cols, rec.NumRows())
+}
+
+func transformArray(arr arrow.Array) arrow.Array {
+	if plainWrite(arr.DataType()) {
+		return arr
+	}
```

Comment:
Why wrap this single boolean expression in a function why not just directly call:
```suggestion
	if arrow.TypeEqual(arr.DataType(), transformTypeForWriting(arr.DataType())) {
		return arr
	}
```

Reply:
Merged this + removed in dafa00461042b75497856cabe705d2bc3c9d535e